### DISCUSSION TITLE
You beat me to it!

### DISCUSSION THREAD

**Discussion by eyelidlessness** [RobinTail/express-zod-api#253] [General]
This project looks **remarkably similar** to one I build internally for a former employer, which I was unfortunately unable to open source before I left. Building a new one from scratch has been on my long list of eventual/future side projects, but I’m quite glad to see someone else arrive at a similar solution!

I’m especially happy to see you’ve also addressed type-safe middlewares and OpenAPI documentation, and that this solution doesn’t depend on decorators. Your interface is very slightly different from the one I arrived at, but spiritually almost identical.

All of which is to say, this looks awesome! I’ll be happy to contribute when I have some cycles to spare.

**Comment by RobinTail** [comment]
Thank you for the feedback @eyelidlessness 
I'm glad that the library meets your high standards.
_I don't like decorators too_ 😆
